src/mon/OSDMonitor.cc

```diff
@@ -5072,6 +5072,22 @@ void OSDMonitor::get_pools_health(
           detail->push_back(s);
       }
     }
+
+    if (mon->monmap->get_required_features().contains_all(
+          ceph::features::mon::FEATURE_LUMINOUS) &&
+        sum.num_objects > 0 && pool.application_metadata.empty()) {
+      stringstream ss;
+      ss << "application not enabled on pool '" << pool_name << "'";
+
+      summary.push_back({HEALTH_WARN, ss.str()});
+      if (detail) {
+        ss << "; "
+           << "use 'ceph osd pool application enable <pool-name> <app-name>', "
+           << "where <app-name> is 'cephfs', 'rbd', 'rgw', or freeform for "
+           << "custom applications.";
+        detail->push_back({HEALTH_WARN, ss.str()});
+      }
+    }
```

Comment:
We definitely need to only do this warning for objects which contain data. Otherwise you'll get a warning as soon as you create a pool.